sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRowToColumnarExec.scala

```diff
@@ -289,6 +289,32 @@ private object GpuRowToColumnConverter {
     }
   }
 
+  private class DecimalConverter(precision: Int, scale: Int) extends TypeConverter {
+    override def append(
+        row: SpecializedGetters,
+        column: Int,
+        builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double = {
+      if (row.isNullAt(column)) {
+        builder.appendNull()
+      } else {
+        new NotNullDecimalConverter(precision, scale).append(row, column, builder)
+      }
+      // Infer the storage type via precision, because we can't access DType of builder.
+      (if (precision > ai.rapids.cudf.DType.DECIMAL32_MAX_PRECISION) 8 else 4) + VALIDITY
+    }
+  }
+
+  private class NotNullDecimalConverter(precision: Int, scale: Int) extends TypeConverter {
+    override def append(
+        row: SpecializedGetters,
+        column: Int,
+        builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double = {
+      builder.append(row.getDecimal(column, precision, scale).toJavaBigDecimal)
+      // Infer the storage type via precision, because we can't access DType of builder.
+      if (precision > ai.rapids.cudf.DType.DECIMAL32_MAX_PRECISION) 8 else 4
```

Comment:
Here too it would be great to avoid conditionals in the data path.  I would prefer it if we either passed in the size or had separate implementations for DECIMAL32 and DECIMAL64